### DISCUSSION TITLE
[Remix] How to customize compiler settings?

### DISCUSSION THREAD

**Discussion by wwwMalcolm** [zkSync-Community-Hub/zksync-developers#785] [Dev Tooling] [answered]
### Environment

Testnet

### Select the Dev Tool you are using

IDE

### Provide the version of the tool (if applicable)

Remix

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Currently I am using Remix and I want to know how do I extract the solidity version that is used when I actually compile? I can see there is an option to choose the zksolc, but not for normal Solidity?

In addition to that, how do I know and configure the settings that was used in my compilation? Example - optimizer settings?

Thank you! 🙏

**Comment by alhenaai** [comment]
In Remix, you can customize and view the compiler settings, including the Solidity version and optimizer settings, through the following steps:

1. **Selecting the Compiler Version:**
   - Open Remix IDE and navigate to the "Solidity Compiler" tab on the left sidebar.
   - Here, you can select the compiler version from a dropdown menu. This dropdown lists all available Solidity versions that you can use for compilation.

2. **Configuring Compiler Settings:**
   - Still in the "Solidity Compiler" tab, you will find a section labeled "Compiler Configuration."
   - Click on the settings icon (a gear icon) to open the detailed settings menu.
   - In this menu, you can configure various settings such as:
     - **Enable Optimization:** Toggle the optimizer and set the number of runs.
     - **EVM Version:** Choose the EVM version to compile for.
     - **Other settings:** Such as metadata, libraries, etc.

3. **Extracting the Solidity Version Used:**
   - The version of Solidity used for compilation is the one you select in the dropdown menu in the "Solidity Compiler" tab.

For using `zksolc` in Remix, if it's available as an option, you would select it similarly to how you select the Solidity version. However, specific settings related to `zksolc` might not be directly configurable in Remix as they are in more integrated development environments like Hardhat or Truffle.

For more detailed control over compilation settings and to use advanced features of `zksolc`, consider switching to a local development environment with tools like Hardhat, which supports `zksolc` through plugins. For more information on setting up and using `zksolc` with Hardhat, you can refer to the [ZKsync documentation on the compiler settings](https://docs.zksync.io/build/tooling/hardhat/plugins/hardhat-zksync-solc#configuration).

If you need further assistance or more detailed examples, please let me know!

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by uF4No** [comment, marked as answer]
Hi @wwwMalcolm 

Currently, there're no UI elements for the compiler settings apart from the zksolc version that you mentioned. However, you can find the compiler settings used by checking the JSON file in the `/build-info` folder. You can search `optimizer_settings` for the optimizer, `solc_version` and `solc_zkvm_edition` for the solidity fork compiler version ([more info here](https://docs.zksync.io/build/tooling/hardhat/plugins/hardhat-zksync-solc#zksync-era-solidity-compiler)) and `zk_version` for the zksolc compiler version among other settings.
